tests/MongoDB.Driver.Tests/Specifications/transactions/TransactionTestRunner.cs

```diff
@@ -118,36 +113,32 @@ private void Dispose(bool disposing)
             }
         }
 
-        private ClusterType MapToplogyToClusterType(BsonValue topology)
+        private void Run(BsonDocument shared, BsonDocument test)
         {
-            switch (topology.AsString)
+            if (shared.TryGetValue("runOn", out var runOn) || shared.TryGetElement("topology", out var topology))
             {
-                case "single": return ClusterType.Standalone;
-                case "replicaset": return ClusterType.ReplicaSet;
-                case "sharded": return ClusterType.Sharded;
-                default: throw new ArgumentException("Unknown topology type");
+                var runOnArray =
+                    runOn != null
+                        ? runOn.AsBsonArray
+                        : new BsonArray(new[] { new BsonDocument(topology.Name, topology.Value) });
+
+                RequireServer.Check().RunOn(runOnArray);
+                CheckTransactionFeature();
```

Comment:
Should the call to CheckTransactionFeature be outside the if statement?

Reply:
I thought so too, but in the previous implementation this method was under the `topology` condition:
    if (shared.TryGetValue("topology", out var topology))
So, I just repeated the previous implementation.
However, I think, previously it was done incorrectly. `topology` is an optional key, unlike checking on whether a feature is supported or not. So, probably it should be validated anyway.
Done